src/backend/gporca/libgpopt/src/xforms/CXformUtils.cpp

```diff
@@ -1342,47 +1342,26 @@ CXformUtils::PexprLogicalDMLOverProject(CMemoryPool *mp,
 	CColRef *pcrAction = NULL;
 	CColRef *pcrOid = NULL;
 
-	if (ptabdesc->IsPartitioned())
+	CExpressionArray *pdrgpexprProjected = GPOS_NEW(mp) CExpressionArray(mp);
+	// generate one project node with two new columns: action, oid (based on the traceflag)
+	pdrgpexprProjected->Append(CUtils::PexprScalarConstInt4(mp, val));
+
+	BOOL fGeneratePartOid = CUtils::FGeneratePartOid(ptabdesc->MDId());
+	if (fGeneratePartOid)
 	{
-		// generate a PartitionSelector node which generates OIDs, then add a project
-		// on top of that to add the action column
-		CExpression *pexprSelector = PexprLogicalPartitionSelector(
-			mp, ptabdesc, colref_array, pexprChild);
-		if (CUtils::FGeneratePartOid(ptabdesc->MDId()))
-		{
-			pcrOid = CLogicalPartitionSelector::PopConvert(pexprSelector->Pop())
-						 ->PcrOid();
-		}
-		pexprProject = CUtils::PexprAddProjection(
-			mp, pexprSelector, CUtils::PexprScalarConstInt4(mp, val));
-		CExpression *pexprPrL = (*pexprProject)[1];
-		pcrAction = CUtils::PcrFromProjElem((*pexprPrL)[0]);
+		OID oidTable = CMDIdGPDB::CastMdid(rel_mdid)->Oid();
+		pdrgpexprProjected->Append(CUtils::PexprScalarConstOid(mp, oidTable));
```

Comment:
This branch where we inject target tableoid into projection list is actual for target non-partitioned tables. Or in case of gpdb this fragment in general have lost the sense due to commit https://github.com/greenplum-db/gpdb/commit/2c139497eadc7cc00d9a178980b865b574663686 (or parent PR in gporca https://github.com/greenplum-db/gporca/pull/186)